### ISSUE TITLE
"Dialog" not announced if the modal is not part of the initial DOM

### ISSUE THREAD

**Issue by mlegait** [nvaccess/nvda#11123]
### Steps to reproduce:
1. Open Firefox or Chrome
2. Browse https://codepen.io/mlegait/pen/eYpEWpz
3. With the keyboard, focus the button "Open Modal"
4. Press "Enter"

### Actual behavior:

"Hello" (aria-labelledby text) is announced.

### Expected behavior:

"Hello dialog" (aria-labelledby text + "dialog") should be announced. Otherwise, the user doesn't know that a modal as been opened.

#### NVDA installed/portable/running from source: 
Installed

#### NVDA version: 
2019.3.1

#### Windows version: 
10 Enterprise

#### Name and version of other software in use when reproducing the issue: 
Firefox 75.0 or Chrome 81.0.4044.129

#### Does the issue still occur after restarting your computer? 
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

#### If addons are disabled, is your problem still occuring?
No addons are installed.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No

#### Additionnal information
At first I opened this issue: https://github.com/reactjs/react-modal/issues/804
But after some investigation with the maintainers of react-modal, it seems that it could be because the modal is not part of the initial DOM. What do you think?

**Comment by Adriani90** [comment]
I cannot reproduce this with NVDA alpha-20161,4e580290 and firefox 76 or Chrome 81.
Could you please thest with these versions?
For me when I press enter on the button from browse mode or focus mode, NVDA reports "hello dialog, heading level 1 hello". If I enable "read page after load in browse mode settings, NVDA reports the whole content of the dialog.
But in any case, it reports hello dialog first, and then the content.
Can you confirm?

cc: @MarcoZehe can you reproduce this in Firefox?

**Comment by MarcoZehe** [comment]
Yes, I can reproduce this. When focus moves into the dialog and onto the document content, the fact that focus has traversed into an actual dialog is not announced. However, object navigator shows the parent of the heading to be a dialog with name "hello". NVDA should have announced that. CC @jcsteh

**Comment by jcsteh** [comment]
This is probably a duplicate; see
https://github.com/nvaccess/nvda/issues/9135#issuecomment-454567979

**Comment by mlegait** [comment]
I didn't see the original bug before opening this one, I'm sorry. So yes, it seems it's a duplicate.

**Comment by Adriani90** [comment]
Closing as duplicate
